yt_dlp/extractor/youtube.py

```diff
@@ -283,12 +284,24 @@ def _real_initialize(self):
     _YT_INITIAL_PLAYER_RESPONSE_RE = r'ytInitialPlayerResponse\s*=\s*({.+?})\s*;'
     _YT_INITIAL_BOUNDARY_RE = r'(?:var\s+meta|</script|\n)'
 
+    def _generate_sapisidhash_header(self):
+        sapisid_cookie = try_get(self._downloader.cookiejar._cookies, lambda x: x['.youtube.com']['/']['SAPISID'])
+        if not hasattr(sapisid_cookie, 'value'):
```

Comment:
`if not hasattr(sapisid_cookie, 'value'):`
tbh this could probably just be replaced by a None check, as its either a NoneType or Cookie object